### PR TITLE
E UI 4650 specify hering stage display story

### PR DESCRIPTION
**Pull request by waynepHMCTS**
### JIRA link (if applicable) ###

https://tools.hmcts.net/jira/browse/EUI-4650

### Change description ###
hearing stage options displayed based on service

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x ] No
```

### CODE REVIEW

File: src/hearings/hearings.module.ts [hunk 1]
```diff
@@ -26,7 +27,7 @@ import { effects, reducers } from './store';
     ReactiveFormsModule,
     ExuiCommonLibModule,
   ],
-  declarations: [...fromContainers.containers],
+  declarations: [...fromContainers.containers, HearingStageComponent],
```
Comment (rayliangatos) [inline src/hearings/hearings.module.ts:30]: can this put into ...fromContainers.containers?

File: src/hearings/containers/request-hearing/hearing-stage/hearing-stage.component.html [hunk 2]
```diff
@@ -0,0 +1,27 @@
+<div class="govuk-width-container">     
+  <main class="govuk-main-wrapper govuk-main-wrapper--auto-spacing" id="main-content" role="main">        
+    <div class="govuk-grid-row">
+      <div class="govuk-grid-column-two-thirds">
+        <form class="form"  [formGroup]="stageForm">
+          <span class="govuk-caption-l">Request a hearing for {{ fullName }} vs DWP</span>
```
Comment (rayliangatos) [inline src/hearings/containers/request-hearing/hearing-stage/hearing-stage.component.html:6]: this should use the generic component 
    exui-hearing-parties-title
```suggestion
          <exui-hearing-parties-title></exui-hearing-parties-title>
```
refer to other components please.

File: src/hearings/containers/request-hearing/hearing-stage/hearing-stage.component.ts [hunk 3]
```diff
@@ -0,0 +1,54 @@
+import { AfterViewInit, Component, OnDestroy, OnInit, QueryList, ViewChildren } from '@angular/core';
+import { FormBuilder, FormGroup } from '@angular/forms';
+import { ActivatedRoute } from '@angular/router';
+import { select, Store } from '@ngrx/store';
+import { Actions } from 'api/hearings/models/hearings.enum';
+import { Observable, Subscription } from 'rxjs';
+import { RefDataModel } from 'src/hearings/models/refData.model';
```
Comment (rayliangatos) [inline src/hearings/containers/request-hearing/hearing-stage/hearing-stage.component.ts:7]: src

File: src/hearings/hearings.routes.ts [hunk 4]
```diff
@@ -72,6 +73,16 @@ export const ROUTES: Routes = [
           title: 'HMCTS Manage cases | Request Hearing | Date Priority Hearing'
         }
       },
+      {
+        path: 'hearing-stage',
+        resolve: { hearingStages: PriorityResolver },
```
Comment (rayliangatos) [inline src/hearings/hearings.routes.ts:78]: if this PriorityResolver can be renamed to RefDataResolver

File: src/hearings/resolvers/priority/priority.resolve.ts [hunk 5]
```diff
@@ -19,15 +19,16 @@ export class PriorityResolver implements Resolve<RefDataModel[]> {
...
     return this.getServiceId$()
       .pipe(
         switchMap(id => {
           return of(
             id ? id : this.serviceId);
         }), take(1),
         switchMap((serviceId) => {
-          return this.getReferenceData$(serviceId);
+          const category = route.data['category'] ? route.data['category'] as HearingCategory : HearingCategory.Priority;
+          return this.getReferenceData$(serviceId, category);
```
Comment (rayliangatos) [inline src/hearings/resolvers/priority/priority.resolve.ts:31]: After this is changed, I suppose you also need to add the 'data' into the route of DatePriorityHearingComponent.

File: src/hearings/resolvers/priority/priority.resolve.ts [hunk 6]
```diff
@@ -38,8 +39,8 @@ export class PriorityResolver implements Resolve<RefDataModel[]> {
     );
   }
 
-  public getReferenceData$(serviceId): Observable<RefDataModel[]> {
-    return this.hearingsDataService.getRefData(HearingCategory.Priority, serviceId).pipe(
+  public getReferenceData$(serviceId, category: HearingCategory = HearingCategory.Priority): Observable<RefDataModel[]> {
```
Comment (rayliangatos) [inline src/hearings/resolvers/priority/priority.resolve.ts:42]: doesn't quite make sense to set a default value here, it should be passed by individual route config.

**Review by rayliangatos — COMMENTED** [review]
And unit test is failed

File: src/hearings/containers/request-hearing/hearing-stage/hearing-stage.component.html (hunk 2 above)
Comment (waynepHMCTS) [reply src/hearings/containers/request-hearing/hearing-stage/hearing-stage.component.html:6]: This not part off the storey

Comment (waynepHMCTS) [reply src/hearings/containers/request-hearing/hearing-stage/hearing-stage.component.html:6]: @rayliangatos this is not part off the story

File: src/hearings/resolvers/priority/priority.resolve.ts (hunk 6 above)
Comment (waynepHMCTS) [reply src/hearings/resolvers/priority/priority.resolve.ts:42]: This is there so it does not break priority

Comment (waynepHMCTS) [reply src/hearings/resolvers/priority/priority.resolve.ts:42]: So I will remove this and force the priority to have data category passed ok

File: src/hearings/containers/request-hearing/hearing-stage/hearing-stage.component.html (hunk 2 above)
Comment (waynepHMCTS) [reply src/hearings/containers/request-hearing/hearing-stage/hearing-stage.component.html:8]: This done by it has not marked it @rayliangatos

File: src/hearings/resolvers/priority/priority.resolve.ts (hunk 5 above)
Comment (waynepHMCTS) [reply src/hearings/resolvers/priority/priority.resolve.ts:31]: This is done but not marked it @rayliangatos

File: src/hearings/containers/request-hearing/hearing-stage/hearing-stage.component.html (hunk 2 above)
Comment (rayliangatos) [reply src/hearings/containers/request-hearing/hearing-stage/hearing-stage.component.html:8]: we need to use a generic component to render all the titles. So you just need to change what I suggested.

**Review by rayliangatos — APPROVED** [review]

File: src/hearings/containers/request-hearing/date-priority-hearing/date-priority-hearing.component.spec.ts [hunk 7]
```diff
@@ -18,7 +18,7 @@ class MockHearingPartiesComponent {
   @Input() public error: ErrorMessage;
 }
 
-describe('DatePriorityHearingComponent', () => {
+fdescribe('DatePriorityHearingComponent', () => {
```
Comment (rayliangatos) [inline src/hearings/containers/request-hearing/date-priority-hearing/date-priority-hearing.component.spec.ts:21]: fdescibe

**Review by Kartheek-S — APPROVED** [review]

**Pull request merged**